File: backend/models/meetup.go
package models

import (
	"time"
)

type Interaction struct {
	ID              uint      `json:"id"`
	Date            time.Time `json:"date"`
	UserID          uint      `json:"user_id"`
	FriendID        uint      `json:"friend_id"`
	InteractionType string    `json:"interaction_type"`
	Attendees       []Friend  `json:"attendees"`
	Name            string    `json:"name"`
	Location        string    `json:"location"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type InteractionAttendee struct {
	ID            uint      `json:"id"`
	IntearctionID uint      `json:"interaction_id"`
	FriendID      uint      `json:"friend_id"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: frontend/src/types/models/Meetup.ts
import type { Friend } from "./Friend";
import type { User } from "./User";

export type Meetup = {
    id: number;
    date: Date;
    location?: string;
    name?: string;
    organizerId: User["id"];
    createdAt?: Date;
    updatedAt?: Date;
}

export type MeetupAttendee = {
    id: number;
    meetupId: Meetup["id"];
    friendId: Friend["id"];
    createdAt?: Date;
    updatedAt?: Date;
}

export type Interaction = {
    id: number;
    date: string;
    userId: User["id"];
    friendId: Friend["id"];
    interactionType?: string;
    createdAt?: Date;
    updatedAt?: Date;
}
